README.md

```diff
@@ -1,16 +1,659 @@
+
+type Action enum {
+  using  URI 
+  do     Ability
+  inputs Any
+} 
+```
+
+## 3.3 JSON Examples
+
+### 3.3.1 Run All
+
+ ``` js
+{
+  "ucan/invoke": {
+    "v": "0.1.0",
+    "prf": ["bafkreifzjut3te2nhyekklss27nh3k72ysco7y32koao5eei66wof36n5e"],
+    "nnc": "abcdef",
+    "run": "*", // Explicitly "run all"
+    "ext": null
+  },
+  "sig": "bdNVZn_uTrQ8bgq5LocO2y3gqIyuEtvYWRUH9YT-SRK6v_SX8bjt-VZ9JIPVTdxkWb6nhVKBt6JGpgnjABpOCA"
+}
+```
+
+### 3.3.2 Promise Pipelines
+
+Promise pipelines are handled in more detail in [section 5](#5-promise-pipelining). In brief, they enable taking the output of an action that has not yet run as the input to another action. Here is a simple example:
+
+``` js
+{
+  "ucan/invoke": {
+    "v": "0.1.0",
+    "prf": ["bafkreifzjut3te2nhyekklss27nh3k72ysco7y32koao5eei66wof36n5e"],
+    "nnc": "02468",
+    "ext": null,
+    "run": {
+      "notify-bob": {
+        "using": "mailto://alice@example.com",
+        "do": "msg/send",
+        "inputs": [
+          {
+            "to": "bob@example.com",
+            "subject": "DNSLink for example.com",
+            "body": "Hello Bob!"
+          }
+        ]
+      },
+      "log-as-done": {
+        "using": "https://example.com/report"
+        "do": "crud/update"
+        "inputs": {
+          "from": "mailto://alice@exmaple.com",
+          "to": ["bob@exmaple.com"],
+          "event": "email-notification",
+          "value": {"ucan/promise": ["/", "notify-bob"]} // Pipelined promise
+        }
+      }
+    }
+  },
+  "sig": "5vNn4--uTeGk_vayyPuNTYJ71Yr2nWkc6AkTv1QPWSgetpsu8SHegWoDakPVTdxkWb6nhVKAz6JdpgnjABppC7"
+}
+```
+
+# 4 Receipt
+
+An invocation receipt is an attestation of the output of an invocation. A receipt MUST be signed by the executor (the `aud` of the associated UCAN).
+
+Note that this does not guarantee correctness of the result! The statement's veracity MUST be only understood as an attestation from the executor.
+
+## 4.1 Fields
+
+| Field | Type          | Description                                                                                      | Required | Default |
+|-------|---------------|--------------------------------------------------------------------------------------------------|----------|---------|
+| `inv` | `&Invocation` | CID of the Invocation that generated this response                                               | Yes      |         |
+| `rlt` | `{CID: Any}`  | The results of each call, indexed by the CID of the `dag-cbor` encoded [Action](#32-ipld-schema) | Yes      |         |
+| `v`   | `SemVer`      | SemVer of the UCAN invocation object schema                                                      | Yes      |         |
+| `nnc` | `String`      | A unique nonce, to distinguish each receipt                                                      | Yes      |         |
+| `ext` | `Any`         | Non-normative extended fields                                                                    | No       | `null`  |
```

Comment:
Is this just metadata ?

Reply:
Yeah, you can use it for metadata, or for people to extend the spec with extra fields as needed. I added some prose the explain.